apps/preview/next/pages/showcases/MegaMenu/OpenOnClick.tsx

```diff
@@ -0,0 +1,263 @@
+import { ShowcasePageLayout } from '../../showcases';
+// #region source
+import {
+  SfIconShoppingCart,
+  SfIconFavorite,
+  SfIconPerson,
+  SfIconMenu,
+  SfIconChevronRight,
+  SfIconClose,
+  SfButton,
+  SfDrawer,
+  SfListItem,
+} from '@storefront-ui/react';
+import { useState, useRef } from 'react';
+import { CSSTransition } from 'react-transition-group';
+import classNames from 'classnames';
+import brandLogo from '@assets/vsf_logo_white.svg';
+import brandLogoSign from '@assets/vsf_logo_sign_white.svg';
+import sneakers from '@assets/sneakers-2.png';
+
+export default function OpenOnClick() {
+  const [open, setOpen] = useState(false);
```

Comment:
Please use [`useDisclosure`](https://docs.storefrontui.io/v2/react/hooks/useDisclosure.html) hook to handle this state.